### DISCUSSION TITLE
Is the format of FauxPilot actually compatible with the VSCode CoPilot Extension?

### DISCUSSION THREAD

**Discussion by romainr** [moyix/fauxpilot#163] [General]
Hello,

This is an awesome project I pointed VSCode to a local server returning data like
https://github.com/fauxpilot/fauxpilot/blob/main/documentation/client.md#api

```
{
  "choices": [
    {
      "finish_reason": "stop",
      "index": 0,
      "logprobs": null,
      "text": "() {\n    return \"Hello, World!\";\n}"
    }
  ],
  "created": 1659492191,
  "id": "cmpl-6hqu8Rcaq25078IHNJNVooU4xLY6w",
  "model": "codegen",
  "object": "text_completion",
  "usage": {
    "completion_tokens": 15,
    "prompt_tokens": 2,
    "total_tokens": 17
  }
}
```

but never got anything displayed.

And it worked withe FauxPilot VSCode extension.

Similar thread:
https://github.com/fauxpilot/fauxpilot/discussions/130

**Comment by thakkarparth007** [comment]
Hi @romainr, to be clear, did you point the vscode to the fauxpilot server or some other server returning the data you showed? The fauxpilot server sends such data with slightly different format:

`data: <jsonobject in one line>`
`<...more such lines...>`
`data: END`

The thread you pointed to was an older one and I think I'd fixed that in a later commit. Let me know if this happens while pointing to the fauxpilot server.
